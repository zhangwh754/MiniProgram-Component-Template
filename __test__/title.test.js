const path = require('path')
const simulate = require('miniprogram-simulate')


function sleep() {
    // 返回一个 Promise，等待 5 秒
    return new Promise((resolve) => {
        setTimeout(() => {
            resolve();
        }, 300);
    });
}

test('Title Component render', async () => {
    const title = '你好世界'
    const id = simulate.load(path.join(process.cwd(), './dist/components/MTitle/index')) // 加载自定义组件，返回组件 id
    const container = simulate.render(id, { title }) // 使用 id 渲染自定义组件，返回组件封装实例

    container.attach(document.createElement('parent-wrapper'))

    expect(container.toJSON()).toMatchSnapshot('initial')

    await sleep()
    container.setData({ title: 'Hello World' })
    expect(container.toJSON()).toMatchSnapshot('updated');
})
