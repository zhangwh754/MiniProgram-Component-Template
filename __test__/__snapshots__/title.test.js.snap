// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component render: initial 1`] = `
<main>
  <wx-view
    class="title"
  >
    你好世界
  </wx-view>
</main>
`;

exports[`Title Component render: updated 1`] = `
<main>
  <wx-view
    class="title"
  >
    Hello World
  </wx-view>
</main>
`;
